_episodes/05-preprocessor.md

```diff
@@ -0,0 +1,287 @@
+>
+>> ## Recipe
+>>
+>>~~~YAML
+>>
+>> datasets:
+>>   - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical,
+>>      ensemble: r1i1p1f2} #single dataset as an example
+>> 
+>> preprocessors:
+>>   prep_map: #preprocessor to just regrid data
+>>     #fill preprocessor details here
+>> 
+>>   prep_map_land: #preprocessor to mask grid cells and then regrid
+>>     #fill preprocessor details here including ordering
+>> 
+>> diagnostics:
+>>   # --------------------------------------------------      
+>>   # Two Simple diagnostics that illustrate the use of
+>>   # different preprocessors
+>>   # --------------------------------------------------
+>>  diag_simple_plot:
+>>    description: # preprocess a variable for a simple 2D plot
+>>    variables:
+>>      # put your variable of choice here
+>>      # apply the first preprocessor i.e. name your preprocessor
+>>      # edit the following 4 lines for mip, grid and time 
+>>      # based on your variable choice
+>>      mip: Amon
+>>      grid: gn #can change for variables from the same model
+>>      start_year: 1970 
+>>      end_year: 2000
+>>    scripts: null #no scripts called
+>>  diag_land_only_plot: #second diagnostic
+>>    description: #preprocess a variable for a 2D land only plot
+>>    variables:
+>>      # include  a variable and information 
+>>      # as in the previous diagnostic and 
+>>      # include your second preprocessor (masking and regridding)
+>>    scripts: null # no scripts
+>>~~~
+>>{: .source}
+>{: .solution}
+>
+>> ## Solution: 
+>> 
+>> Here is one possible way to use two different preprocessors including a 
+>> group of preprocessors on different variables.
+>> 
+>>~~~YAML
+>>
+>> datasets:
+>>   - {dataset: UKESM1-0-LL, project: CMIP6, exp: historical,
+>>      ensemble: r1i1p1f2} #single dataset as an example
+>> 
+>> preprocessors:
+>>   prep_map:
+>>     regrid:    #apply the preprocessor to regrid
+>>       target_grid: 1x1 # target resolution
+>>       scheme: linear  #how to interpolate for regridding
+>> 
+>>   prep_map_land:
+>>     custom_order: true #ensure order follows a given
+>>     mask_landsea:    #apply a mask
+>>       mask_out: sea   #mask out sea grid cells
+>>     regrid:    # now apply the preprocessor to regrid
+>>       target_grid: 1x1 # target resolution
+>>       scheme: linear  #how to interpolate for regridding
+>> 
+>> diagnostics:
+>>   # --------------------------------------------------      
+>>   # Two Simple diagnostics that illustrate the use of
+>>   # different preprocessors
+>>   # --------------------------------------------------
+>>   diag_simple_plot:
+>>     description: # preprocess a variable for a simple 2D plot
+>>     variables:
+>>       tas:  # surface temperature
+>>         preprocessor: prep_map
+>>         mip: Amon
+>>         grid: gn #can change for variables from the same model
+>>         start_year: 1970 
+>>         end_year: 2000
+>>         additional_datasets:
```

Comment:
"additional_datasets" were not mentioned in the script above. Maybe take it out here as well?